lib/core/src/Cardano/Wallet/Network.hs

```diff
@@ -25,35 +25,43 @@ module Cardano.Wallet.Network
 import Prelude
 
 import Cardano.BM.Trace
-    ( Trace, logDebug, logError, logInfo, logNotice )
+    ( Trace, logDebug, logError, logInfo, logNotice, logWarning )
 import Cardano.Wallet.Primitive.Model
     ( BlockchainParameters (..) )
 import Cardano.Wallet.Primitive.Types
     ( BlockHeader (..), Hash (..), TxWitness )
 import Control.Concurrent
     ( threadDelay )
 import Control.Exception
-    ( Exception (..) )
+    ( Exception (..), SomeException )
+import Control.Monad
+    ( when )
+import Control.Monad.Catch
+    ( Handler )
 import Control.Monad.IO.Class
     ( liftIO )
 import Control.Monad.Trans.Except
     ( ExceptT, runExceptT )
 import Control.Retry
     ( RetryPolicyM
+    , RetryStatus (..)
     , exponentialBackoff
     , limitRetriesByCumulativeDelay
+    , logRetries
+    , recovering
     , retrying
     )
 import Data.Text
     ( Text )
 import Fmt
-    ( (+|), (+||), (|+), (||+) )
```

Comment:
I've removed all of these operators that are error-prone and a pain to use in the end. `pretty` and `<>` is much simpler.